src/Functions/formatDateTime.cpp

```diff
@@ -936,6 +994,11 @@ class FunctionFormatDateTimeImpl : public IFunction
                         out_template += "00";
                         break;
 
+                    // Full month name [January...December]
```

Comment:
Semantics of "M" change (l. 983ff. on the left side) which technically makes this PR backwards-incompatible. This is okay but we should mark it as such - I'll add the right tags. It will then show up in the backward-incompatible section of the monthly changelog.